### ISSUE TITLE
Using hand-written loops instead of STL copy() and ostream_iterator<>

### ISSUE THREAD

**Issue by nabijaczleweli** [Anmol-Singh-Jaggi/Pretty-print#2]
As in the title.
Consider this (yours) code:

``` c++
template<typename T1>
std::ostream& operator<<(std::ostream& out, const std::vector<T1>& object) {
  out << "[";
  for(const auto& it : object) {
    out << it << ", ";
  }
  if(object.size()) {
    out << "\b\b";
  }
  out << "]";
  return out;
}
```

Now consider this one:

``` c++
template<typename T1>
std::ostream& operator<<(std::ostream& out, const std::vector<T1>& object) {
  out << "[";
  copy(object.begin(), object.end(), std::ostream_iterator<T1>(out, ", "));
  if(object.size()) {
    out << "\b\b";
  }
  out << "]";
  return out;
}
```

As you can see, the second one is much cleaner, and uses `std::copy()`, and is therefore recommended and more performant.

**Comment by nabijaczleweli** [comment]
Closing for...?

**Comment by Anmol-Singh-Jaggi** [comment]
Ummm, closing because I have noted the advice and going to soon commit the changes to the new version. 
I dont know much about Github and I apologize if the closing feature is used for something else.

**Comment by nabijaczleweli** [comment]
Oh, cool. Just as a future advice, you can close issues directly from commits (just add "Closes #N" to the commit message (as in: "Added vodka and salt. Closes #2")).
It also makes it easier to see manage which commits change and close what.

**Comment by Anmol-Singh-Jaggi** [comment]
Ahhh...nice idea. If only you would have told me this earlier; I have already made the commit with the changes you mentioned.
